### PR TITLE
Option to render learned geometry in video rollouts

### PR DESCRIPTION
**Pull request by ebianchi**
### Contents
The main aim of this PR is to add in the ability to choose whether rollout videos render the predicted trajectories with the original geometry or with the current learned geometry.  This decision can be toggled via the flag `update_geometry_in_videos` in the configuration class `SupervisedLearningExperimentConfig`.

Tangential to this main aim, the following other minor changes are also incorporated:
- Some hard-coded file/directory locations (a remnant of PR #3 ) are now implemented in a slightly less hacky way.  This change is included in this PR because this added functionality requires exporting an updated URDF for every toss video generated.  The exported URDF file path from this PR is created in a similar way as the exported video gif file from the prior PR, so it made sense to handle both in a consistent way.
- The color of the true system in the rollout videos is now slightly more transparent.  This helped see both the ground truth and the learned system trajectories especially when the learned system geometry is much smaller than the ground truth geometry.
- The example script `examples/contactnets_simple.py` features setting `update_geometry_in_videos = True` so reviewers can demo this PR feature.

### Call for comments
In this implementation, the `MultibodyLearnableSystem` extension of the `System` class now has an additional (optional) argument for its `summary` method.  This argument is the flag for whether to use original or learned geometry in the generated rollout videos.  As originally created, this PR does _not_ include any edits to the original `System` class or any other extensions of it to maintain a consistent set of arguments for `summary()`.  This can be added in if desired, and I welcome feedback on this point in particular.

### Examples
The following two gifs are of the same ground truth trajectory (in red) with different predicted trajectories (in blue) as the experiment updates the believed object properties.  The geometric updates can be observed in these toss videos.

toss_learned_geom0
toss_learned_geom1

### CODE REVIEW

File: dair_pll/multibody_learnable_system.py [hunk 1]
```diff
@@ -114,20 +115,32 @@ def get_visualization_system(self) -> DrakeSystem:
...
 
+        Args:
+            new_geometry: Whether or not to use learned geometry.
+
         Returns:
             New :py:class:`DrakeSystem` with doubled state and repainted
             elements.
         """
-        if not self.visualization_system:
+        if new_geometry or not self.visualization_system:
```
Comment (mshalm) [inline dair_pll/multibody_learnable_system.py:127]: Reason for `new_geometry` -> regenerate visualization system is non obvious enough that I think it deserves a comment

File: dair_pll/vis_utils.py [hunk 2]
```diff
@@ -39,6 +39,7 @@
 
 def generate_visualization_system(
         base_system: DrakeSystem,
+        learned_system: DrakeSystem = None,
```
Comment (mshalm) [inline dair_pll/vis_utils.py]: type hint should be `Optional[DrakeSystem]` instead of `DrakeSystem`

File: dair_pll/vis_utils.py [hunk 3]
```diff
@@ -49,19 +50,37 @@ def generate_visualization_system(
...
 
     Returns:
         New ``DrakeSystem`` with doubled state and repainted elements.
     """
-    # Creates duplicated system
+    # Start with true base system.
     double_urdfs = deepcopy(base_system.urdfs)
     double_urdfs.update({
-        (k + LEARNED_TAG): v for k, v in base_system.urdfs.items()
+        k: v for k, v in base_system.urdfs.items()
```
Comment (mshalm) [inline dair_pll/vis_utils.py]: this appears to be a no-op

File: dair_pll/vis_utils.py [hunk 4]
```diff
@@ -49,19 +50,37 @@ def generate_visualization_system(
...
+    if learned_system is None:
+        double_urdfs.update({
+            (k + LEARNED_TAG): v for k, v in base_system.urdfs.items()
+        })
+
+    else:
+        double_urdfs.update({
+            (k + LEARNED_TAG): v for k, v in learned_system.urdfs.items()
+        })
+
```
Comment (mshalm) [inline dair_pll/vis_utils.py]: copy-paste code block -- could be compressed as something like
```
system_to_add = base_system if not learned_system else learned_system
double_urdfs.update( ...
... system_to_add.urdfs.items()
)
```

File: dair_pll/drake_utils.py [hunk 5]
```diff
@@ -56,9 +56,8 @@
 CAM_FOV = np.pi/5
 VIDEO_PIXELS = [480, 640]
 FPS = 30
-SENSOR_POSE = RigidTransform(RollPitchYaw([-np.pi/2, 0, np.pi/2]), [1, 0, 0.2])
-EXP_NAME = file_utils.EXP_NAME
-VIDEO_FILENAME = file_utils.VIDEO_FILENAME
+SENSOR_POSE = RigidTransform(RollPitchYaw([-np.pi/2, 0, np.pi/2]), [2, 0, 0.2])
```
Comment (mshalm) [inline dair_pll/drake_utils.py:62]: This is an issue from my code earlier, not this PR, but the hard-coded camera pose deserves a `TODO` comment to dynamically set it based on the volume swept by the actual trajectory. I think this PR makes sense as a time to add it

File: dair_pll/drake_utils.py [hunk 6]
```diff
@@ -56,9 +56,8 @@
...
+VIDEO_FILENAME = file_utils.get_experiment_video_filename()
```
Comment (mshalm) [inline dair_pll/drake_utils.py]: Stylistically, probably not much of a reason to create a static variable that's just a single call to a member of `file_utils`

**Review by mshalm — CHANGES_REQUESTED** [review]
Overall clean implementation of learned geometry visualization.

This PR has unfortunately run up against some prior lack of thoughtfulness on my part on file system management. Namely, `dair_pll` was designed to always manage files in units of `Study`, not `Experiment`. However, as the use case in this PR has brought to light, obviously it makes sense to have file storage for individual experiments that's managed well. There likely should be some encapsulation of an "experiment directory" within the `Experiment` class, probably in the config. The solution for now works fine, but I think an issue should be formed before this PR is merged so we don't lose track of the slight hackiness that's still present.

File: dair_pll/vis_utils.py (hunk 3 above)
Comment (ebianchi) [reply dair_pll/vis_utils.py]: Good catch -- this and the copy-paste code block below were edited versions of lines in my fork that did do distinct and real tasks (namely find a geometrically accurate version of URDF used for initialization), but the edits to keep that other functionality out of this PR rendered these lines mostly not useful.  I removed this line and combined the others as suggested.

**Comment by ebianchi** [comment]
All comments addressed without disagreement.

**Comment by ebianchi** [comment]
I additionally opened an issue #7 as a reminder to address the current hacked way of getting experiment-level storage locations, so that the issue is remembered beyond the closing of this PR.

**Review by mshalm — APPROVED** [review]
LGTM!

**Pull request merged**